### PR TITLE
feat(enhanced websocket): Add `transactionSubscribe` and `accountSubscribe`

### PR DESCRIPTION
**Pull request by 0xBreath**
Added support for [geyser enhanced websockets](https://docs.helius.dev/webhooks-and-websockets/websockets#helius-geyser-enhanced-websockets-beta).

There are two methods each with an example:
- [Transaction subscription](https://github.com/0xBreath/helius-rust-sdk/blob/enhanced-ws/examples/enhanced_websocket_accounts.rs)
- [Account subscription](https://github.com/0xBreath/helius-rust-sdk/blob/enhanced-ws/examples/enhanced_websocket_transactions.rs)

Tests pass, cargo fmt has been run, and all warnings have been cleaned.

### CODE REVIEW

File: src/websocket.rs [hunk 1]
```diff
@@ -0,0 +1,372 @@
+#![allow(dead_code)]
```
Comment (0xIchigo) [inline src/websocket.rs:1]: Can you remove this and clean up the relevant dead code, please?

File: src/websocket.rs [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        protocol::frame::{coding::CloseCode, CloseFrame},
+        Message,
+    },
+    MaybeTlsStream, WebSocketStream,
+};
+use url::Url;
+
+pub const ENHANCED_WEBSOCKET_URL: &str = "wss://atlas-mainnet.helius-rpc.com?api-key=";
+
+pub type EnhancedWebsocketResult<T = ()> = Result<T, HeliusError>;
```
Comment (0xIchigo) [inline src/websocket.rs:42]: This could be cleaned up using the `Result` type alias mentioned earlier

File: src/websocket.rs [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+#[derive(Debug)]
+pub struct EnhancedWebsocket {
+    subscribe_sender: mpsc::UnboundedSender<SubscribeRequestMsg>,
+    request_sender: mpsc::UnboundedSender<RequestMsg>,
+    shutdown_sender: oneshot::Sender<()>,
+    node_version: RwLock<Option<semver::Version>>,
+    ws: JoinHandle<EnhancedWebsocketResult>,
+}
+
+impl EnhancedWebsocket {
```
Comment (0xIchigo) [inline src/websocket.rs:62]: More fleshed-out documentation for these new methods would be greatly appreciated

File: src/websocket.rs [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+        self.request_sender
+            .send(("getVersion".to_string(), Value::Null, response_sender))
+            .map_err(|err| HeliusError::WebsocketClosed(err.to_string()))?;
+        let result = response_receiver
+            .await
+            .map_err(|err| HeliusError::WebsocketClosed(err.to_string()))??;
+        let node_version: RpcVersionInfo = serde_json::from_value(result)?;
+        let node_version =
+            semver::Version::parse(&node_version.solana_core).map_err(|e| HeliusError::EnhancedWebsocket {
+                reason: format!("failed to parse cluster version: {e}"),
```
Comment (0xIchigo) [inline src/websocket.rs:121]: Can you capitalize this please so it reads `"Failed to parse cluster..."`

**Review by 0xIchigo — CHANGES_REQUESTED** [review]
Very based! The code is solid overall, I just have a few nitpicks here and there that could improve its overall quality. Also, I've tested the examples with a few real-world calls and can ensure that it works as intended. If possible, it would be nice if you could write up some test cases for `transactionSubscribe` and `accountSubscribe` in a `test/test_websocket.rs` file

File: .gitignore
[the remark this answers is not in the document]
Comment (0xIchigo) [reply .gitignore:1]: On second thought, disregard this comment lol all good!

File: examples/enhanced_websocket_accounts.rs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+use helius::error::HeliusError;
+use helius::types::Cluster;
+use helius::Helius;
+use solana_sdk::pubkey;
+use tokio_stream::StreamExt;
+
+#[tokio::main]
+async fn main() -> Result<(), HeliusError> {
```
[the remark this answers is not in the document]
Comment (0xBreath) [reply examples/enhanced_websocket_accounts.rs:8]: I copied and pasted from your other tests. Ctrl+F for `Result<(), HeliusError`.
I'll use your result, but also modify your existing tests for consistency.

**Pull request merged**